packages/edit-site/src/components/navigation-sidebar/navigation-panel/main-back-button.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalUseSlot as useSlot,
+	createSlotFill,
+} from '@wordpress/components';
+
+const slotName = '__experimentalNavigationPanelMainBackButton';
+
+const { Fill, Slot: NavigationPanelMainBackButtonSlot } = createSlotFill(
+	slotName
+);
+
+const NavigationPanelMainBackButton = Fill;
+
+const Slot = ( { children } ) => {
+	const slot = useSlot( slotName );
+	const hasFills = Boolean( slot.fills && slot.fills.length );
+
+	if ( ! hasFills ) {
+		return children;
+	}
+
+	return <NavigationPanelMainBackButtonSlot />;
+};
+
+NavigationPanelMainBackButton.Slot = Slot;
+
+export default NavigationPanelMainBackButton;
```

Comment:
What do you think of moving this out of `navigation-panel` and into a separate folder?
E.g. `navigation-sidebar/navigation-main-back-button/index.php`

We are using it with the `NavigationBackButton` component, but there's nothing preventing consumers to use anything else, practically making it decoupled from its `Navigation` context.

Reply:
Im a bit confused how it would be decoupled given we are initializing the slot in this package either way?

Reply:
Fair enough. 🙂